utils/prof_utils.py

```diff
@@ -616,6 +616,126 @@ def est_sn_from_prof(prof_data, period, alpha=3.):
+    -----------
+    prof_data: list
+        A list of floats that contains the pulse profile. Default
+    period: float
+        The pulsar's period in ms
+    verbose: boolean
+        OPTIONAL - Determines whether to return more detailed information. Detailed below. Default: True
+    Returns:
+    --------
+    sn: float
+        The estimated signal to noise ratio
+    u_sn: float
+        The estimated signal to noise ratio's its uncertainty
+    flags: list
+        VERBOSE - a list of flagged data points
+    w_equiv_bins: float
+        VERBOSE - the equivalent width of the profile measured in bins
+    u_w_equiv_bins: float
+        VERBOSE - the uncertaintiy in w_equiv_bins
+    w_equiv_ms: float
+        VERBOSE - the equivalent width of the profile measured in ms
+    u_w_equiv_ms: float
+        VERBOSE - the uncertainty in w_equiv_ms
+    scattering: float
+        VERBOSE: - the scattering width in
+    u_scattering: float
+    scattered: boolean
+        VERBOSE - when true, the profile is highly scattered
+    """
+
+    nbins = len(prof_data)
+    #centre the profile around the max
+    shift = -int(np.argmax(prof_data))+int(nbins)//2
+    prof_data = np.roll(prof_data, shift)
+
+    #find sigma and check if profile is scattered
+    sigma, flags = submit_to_database.sigmaClip(prof_data, alpha=3., tol=0.01, ntrials=100)
+    bot_prof_min = (max(prof_data) - min(prof_data)) * .1 + min(prof_data)
+    scattered=False
+    if (np.nanmin(flags) > bot_prof_min) or ( not np.isnan(flags).any() ):
+        logger.warning("The profile is highly scattered. S/N estimate cannot be calculated")
+        if verbose == True:
+            scattered=True
+            #making a new profile with the only bin being the lowest point
+            prof_min_i = np.argmin(prof_data)
+            flags = []
+            for fi, _ in enumerate(prof_data):
+                if fi == prof_min_i:
+                    flags.append(prof_data[fi])
+                else:
+                    flags.append(np.nan)
+
+            flags = np.array(flags)
+            prof_data -= min(prof_data)
+            #Assuming width is equal to pulsar period because of the scattering
+            w_equiv_ms = period
+            u_w_equiv_ms = period/nbins
+        sn = None
+        u_sn = None
+    else:
+        u_prof = 500. #this is an approximation
+        pulse_width_bins = 0
+        non_pulse_bins = 0
+        p_total = 0.
+        u_p = 0.
+        #work out the above parameters
+        for i, data in enumerate(prof_data):
+            if np.isnan(flags[i]):
+                pulse_width_bins += 1
+                p_total += data
+                u_p = np.sqrt(u_p**2 + u_prof**2)
+            else:
+                non_pulse_bins += 1
+        u_simga = sigma / np.sqrt(2 * non_pulse_bins - 2)
+
+        #now calc S/N
+        sn = max(prof_data)/sigma
+        u_sn = sn * np.sqrt(u_prof/max(prof_data)**2 + (u_simga/sigma)**2)
+
+    if verbose==False:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [comparison to False should be 'if cond is False:' or 'if not cond:'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)